ports/esp32/mphalport.c

```diff
@@ -78,6 +84,9 @@ void check_esp_err(esp_err_t code) {
         o_str->data = (const byte *)esp_err_to_name(code); // esp_err_to_name ret's ptr to const str
         o_str->len = strlen((char *)o_str->data);
         o_str->hash = qstr_compute_hash(o_str->data, o_str->len);
+        #if MICROPY_ERROR_REPORTING > MICROPY_ERROR_REPORTING_NORMAL
+        mp_printf(MP_PYTHON_PRINTER, "Exception in function '%s' at line %d in file '%s'\n", func, line, file);
```

Comment:
In other parts of MicroPython `MICROPY_ERROR_REPORTING_DETAILED` doesn't print to the console it adds additional information into existing output (exception strings, `print(obj)`, etc.)

Suggest that to be consistent with that, it'd be better to create a longer `o_str` that includes the function, line and file information. Then it can become part of the OSError exception itself. This also means that it won't spam the console if an exception is thrown and caught as part of expected control flow.

(For completeness: there is a different flag that does add extra standard output debug logging, `MICROPY_DEBUG_VERBOSE`. However this flag adds a *lot* of output, so it's only useful for certain cases and I think `MICROPY_ERROR_REPORTING_DETAILED` is the better flag to use for this.)

Reply:
I provoke the error by `10+self->mode`
```
check_esp_err(ledc_set_duty(10+self->mode, self->channel, channel_duty));
```
and get exception
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: -258, 0x0102, 'ESP_ERR_INVALID_ARG' in function 'set_duty_u16' at line 342 in file './machine_pwm.c'
```

Reply:
Nice!